ext/pcntl/pcntl.c

```diff
@@ -692,8 +692,17 @@ PHP_FUNCTION(pcntl_signal_get_handler)
 		Z_PARAM_LONG(signo)
 	ZEND_PARSE_PARAMETERS_END();
 
-	if (signo < 1 || signo > 32) {
-		zend_argument_value_error(1, "must be between 1 and 32");
+	// note: max signal on mac is SIGUSR2 (31), no real time signals.
+	static int sigmax = NSIG - 1;
```

Comment:
I think I'd prefer a non-static variable here. Otherwise we have it in a data section that's shared with other threads too.
The compiler will constant-fold and take care of the branch at compile-time anyway.

Reply:
I wish we had constexpr in C :-) but sure will change.

Reply:
> I wish we had constexpr in C :-)

That's in C23 ;) But sadly we don't even have C11 yet in PHP